hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/SparkHoodieIndexFactory.java

```diff
@@ -104,13 +100,4 @@ public static boolean isGlobalIndex(HoodieWriteConfig config) {
         return createIndex(config).isGlobal();
     }
   }
-
-  private static Option<BaseKeyGenerator> getKeyGeneratorForSimpleIndex(HoodieWriteConfig config) {
```

Comment:
All similar logic is moved to `HoodieSparkKeyGeneratorFactory.createVirtualKeyGenerator`